sdk/nodejs/runtime/state.ts

```diff
@@ -31,6 +31,8 @@ const nodeEnvKeys = {
     // this value is not set by the CLI and is controlled via a user set env var unlike the values above
     cacheDynamicProviders: "PULUMI_NODEJS_CACHE_DYNAMIC_PROVIDERS",
     organization: "PULUMI_NODEJS_ORGANIZATION",
+    // this value is not set by the CLI and is controlled via a user set env var unlike the values above
+    throwOutputToString: "PULUMI_NODEJS_THROW_OUTPUT_TOSTRING",
```

Comment:
I'd probably call this something like "PULUMI_ERROR_OUTPUT_STRING" as it would make sense to use the same envvar in other languages.

Further it would make sense to add this to sdk/go/common/env/env.go so it shows up in the CLI's env help.

Reply:
the Go SDK at least already has the JSON marshaler on outputs return an error, e.g. https://github.com/pulumi/pulumi/blob/05f64755af2c578c6fb41a4c1427063b058123f4/sdk/go/pulumi/types.go#L257, so this option may not apply to all languages

Reply:
The python SDK on the other hand also returns a string instead of raising an error:
https://github.com/pulumi/pulumi/blob/master/sdk/python/lib/pulumi/output.py#L725

Reply:
> the Go SDK at least already has the JSON marshaler on outputs return an error, e.g.

That's specific to JSON marshaling. Go doesn't let you call `String()` or `sprintf("%s")` on outputs.

> The python SDK on the other hand also returns a string instead of raising an error:

Yes, same as the node SDK, and the Java and dotnet sdk.

> so this option may not apply to all languages

It won't apply to _all_ languages but it will apply to _most_ languages. It should be named generally, not specific to NODEJS.

Reply:
Thanks for the clarification. I've added a declaration to `sdk/go/common/env/env.go`, should I be doing anything else with that variable?

Reply:
No, just need it there so it can show up in env help.